ONSdigital/ras-rm-performance-tests
# Recreate jmeter tests

This recreates the old JMeter test script in Locust (and removes the old JMeter stuff). The new script does a full data load on installation if the data isn't already available.

**How to test:**

1. Make sure you have a clean environment with `ras trigger <cluster> <namespace>`
2. Change `values.yaml` to point to the correct URLs for the various services
3. From `_infra/helm`, run `helm install locust locust`
4. `kc port-forward <master pod> 8089`
5. Navigate to http://localhost:8089/
6. When all the slaves are showing in the top-right corner (5 by default), run a performance test. I would suggest keeping it to 50 users hatched 10 at a time in sandbox or dev; the cluster isn't beefy enough to handle 150 users and you'll start seeing legitimate failures from Secure Message timing out
7. Watch it go! There shouldn't be any Failures or Exceptions. The test is "done" when `GET /surveys/history` gets called - a user then loops back to the beginning of the test.
